core/commands/partials/vagrant_configurator.rb

```diff
@@ -146,8 +146,8 @@ def bring_up_and_configure(node, logger)
       next unless VagrantService.node_running?(node, logger)
 
       @network_config.add_nodes([node])
-      unless NetworkChecker.resources_available?(@machine_configurator, @network_config[node], logger)
-        @ui.error('Network resources not available!')
+      unless NetworkChecker.resources_available?(MachineConfigurator.new(logger), @network_config[node], logger)
```

Comment:
You must fix the NetworkChecker class, do not create new instance of MachineConfigurator class.